x-pack/plugins/apm/server/lib/service_map/get_service_anomalies.ts

```diff
@@ -39,29 +40,13 @@ export async function getServiceAnomalies({
     throw Boom.notImplemented(ML_ERRORS.ML_NOT_AVAILABLE);
   }
 
-  const mlCapabilities = await ml.mlSystem.mlCapabilities();
-
-  if (!mlCapabilities.mlFeatureEnabledInSpace) {
-    throw Boom.forbidden(ML_ERRORS.ML_NOT_AVAILABLE_IN_SPACE);
-  }
-
-  const mlJobIds = await getMLJobIds(ml.anomalyDetectors, environment);
-
-  if (!mlJobIds.length) {
-    return {
-      mlJobIds: [],
-      serviceAnomalies: {},
-    };
-  }
-
   const params = {
     body: {
       size: 0,
       query: {
         bool: {
           filter: [
             { terms: { result_type: ['model_plot', 'record'] } },
-            { terms: { job_id: mlJobIds } },
```

Comment:
i'm wondering how well this search will perform on a system that has millions of non-apm result records.
would the overall performance be better if the apm job ids where loaded up front and added as a terms filter as before, but then not passing the ids to `mlAnomalySearch` as per this PR to avoid that secondary spaces check

Reply:
I was wondering the same but as @dgieselaar said, we already filter by apm specific values via:
```
by_field_value: [TRANSACTION_REQUEST, TRANSACTION_PAGE_LOAD],
```

This is not guaranteed to scope the results to APM since other apps could have the same values (not very likely though).
To make sure only APM results are returned can we filter by: `app: 'apm'` or something similar?

Even if only APM results are returned, it could be a big response if many APM ML jobs exist. Not sure if that's an actual problem though

Reply:
> To make sure only APM results are returned can we filter by: app: 'apm' or something similar?

The only identifying data in the result document will be the `job_id` .